cpp/src/reductions/segmented_reductions.cpp

```diff
@@ -98,8 +127,33 @@ std::unique_ptr<column> segmented_reduce(column_view const& segmented_values,
                                          rmm::mr::device_memory_resource* mr)
 {
   CUDF_FUNC_RANGE();
-  return detail::segmented_reduce(
-    segmented_values, offsets, agg, output_dtype, null_handling, cudf::default_stream_value, mr);
+  return detail::segmented_reduce(segmented_values,
+                                  offsets,
+                                  agg,
+                                  output_dtype,
+                                  null_handling,
+                                  std::nullopt,
+                                  cudf::default_stream_value,
+                                  mr);
+}
+
+std::unique_ptr<column> segmented_reduce(column_view const& segmented_values,
+                                         device_span<size_type const> offsets,
+                                         segmented_reduce_aggregation const& agg,
+                                         data_type output_dtype,
+                                         null_policy null_handling,
+                                         scalar const& init,
```

Comment:
Similar to above, use `std::optional<std::reference_wrapper<scalar const>> init`.